builtin/providers/aws/resource_aws_api_gateway_api_key_test.go

```diff
@@ -0,0 +1,182 @@
+					testAccCheckAWSAPIGatewayApiKeyAttributes(&conf),
+				),
+			},
+		},
+	})
+}
+
+func testAccCheckAWSAPIGatewayApiKeyAttributes(conf *apigateway.ApiKey) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		if *conf.Id == "" {
+			return fmt.Errorf("empty ID")
+		}
+		if len(conf.StageKeys) != 1 {
+			return fmt.Errorf("Expected one apikey. Got %d", len(conf.StageKeys))
+		}
+		if !strings.Contains(*conf.StageKeys[0], "test") {
+			return fmt.Errorf("Expected apikey for test. Got %q", *conf.StageKeys[0])
+		}
+		return nil
+	}
+}
+
+func testAccCheckAWSAPIGatewayApiKeyExists(n string, res *apigateway.ApiKey) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		rs, ok := s.RootModule().Resources[n]
+		if !ok {
+			return fmt.Errorf("Not found: %s", n)
+		}
+
+		if rs.Primary.ID == "" {
+			return fmt.Errorf("No API Gateway ApiKey ID is set")
+		}
+
+		conn := testAccProvider.Meta().(*AWSClient).apigateway
+
+		req := &apigateway.GetApiKeyInput{
+			ApiKey: aws.String(rs.Primary.ID),
+		}
+		describe, err := conn.GetApiKey(req)
+		if err != nil {
+			return err
+		}
+
+		if *describe.Id != rs.Primary.ID {
+			return fmt.Errorf("APIGateway ApiKey not found")
+		}
+
+		res.Id = describe.Id
+		res.Description = describe.Description
+		res.Enabled = describe.Enabled
+		res.LastUpdatedDate = describe.LastUpdatedDate
+		res.Name = describe.Name
+		res.StageKeys = describe.StageKeys
+
+		return nil
+	}
+}
+
+func testAccCheckAWSAPIGatewayApiKeyDestroy(s *terraform.State) error {
+	conn := testAccProvider.Meta().(*AWSClient).apigateway
+
+	for _, rs := range s.RootModule().Resources {
+		if rs.Type != "aws_api_gateway_api_key" {
+			continue
+		}
+
+		describe, err := conn.GetApiKeys(&apigateway.GetApiKeysInput{})
+
+		if err == nil {
+			if len(describe.Items) != 0 &&
+				*describe.Items[0].Id == rs.Primary.ID {
+				return fmt.Errorf("API Gateway ApiKey still exists")
+			}
+		}
+
+		aws2err, ok := err.(awserr.Error)
+		if !ok {
+			return err
+		}
+		if aws2err.Code() != "NotFoundException" {
+			return err
+		}
+
+		return nil
+	}
+
+	return nil
+}
+
+const testAccAWSAPIGatewayApiKeyConfig = `
+resource "aws_api_gateway_rest_api" "test" {
+  name = "test"
+}
+
+resource "aws_api_gateway_resource" "test" {
+  api_id = "${aws_api_gateway_rest_api.test.id}"
+  parent_resource_id = "${aws_api_gateway_rest_api.test.root_resource_id}"
+  path_part = "test"
+}
+
+resource "aws_api_gateway_method" "test" {
+  api_id = "${aws_api_gateway_rest_api.test.id}"
+  resource_id = "${aws_api_gateway_resource.test.id}"
+  http_method = "GET"
+  authorization = "NONE"
+}
+
+resource "aws_api_gateway_method_response" "error" {
+  api_id = "${aws_api_gateway_rest_api.test.id}"
+  resource_id = "${aws_api_gateway_resource.test.id}"
+  http_method = "${aws_api_gateway_method.test.http_method}"
+  status_code = "400"
+}
+
+resource "aws_api_gateway_integration" "test" {
+  depends_on = ["aws_api_gateway_method.test"]
```

Comment:
This line shouldn't be necessary, the relationship is already defined 3 lines below - `http_method = "${aws_api_gateway_method.test.http_method}"`.